### PR TITLE
modem: hl7800: ensure net_if_down is triggered and not ignored

### PR DESCRIPTION
**Pull request by rerickson1**
Trigger network dropped based on socket error notifications.
This debounces the network state and only triggers
the network interface to go down if the network
really drops or causes socket problems.
This will ensure upper networking stack layers can
cleanup broken sockets properly.

Debounce DNS resolver refresh.
Only trigger DNS resolver refresh if the DNS address changes.

### CODE REVIEW

**Comment by bjarki-trackunit** [comment]
Looks fine :)

**Comment by rerickson1** [comment]
Rebased, no changes compared to before the rebase. @bjarki-trackunit  can you please re-review and signoff?

**Review by bjarki-andreasen — DISMISSED** [review]
ok

**Comment by bjarki-trackunit** [comment]
> Rebased, no changes compared to before the rebase. @bjarki-trackunit can you please re-review and signoff?

Done, I do not have write access though so you still need two more :)

**Review by rlubos — DISMISSED** [review]

File: drivers/modem/hl7800.c [hunk 1]
```diff
@@ -2021,14 +2024,25 @@ static bool on_cmd_atcmdinfo_ipaddr(struct net_buf **buf, uint16_t len)
 	/* store new dns */
 	addr_start = delims[4] + 1;
 	addr_len = delims[5] - addr_start;
+	strncpy(temp_addr_str, addr_start, addr_len);
+	temp_addr_str[addr_len] = 0;
 	if (is_ipv4) {
+		ret = strncmp(temp_addr_str, ictx.dns_v4_string,
+							addr_len);
```
Comment (ycsin) [inline drivers/modem/hl7800.c:2031]: I believe this can be fit into one line?

**Review by ycsin — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**